From 1772826545aaa80e6ccbdb156e88b7b545afe63e Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sat, 18 Jan 2020 22:04:33 +0200
Subject: [PATCH] [SW-8228] arcboot: remove writeb()/readb()

We only acces dwords in configuration space and also in FW sram

Change-Id: I661db64d90fe4faef35ca28eaac6e297cd7e5cbf
---

diff --git a/include/arcboot.h b/include/arcboot.h
index 26fca48..364c1bc 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -44,9 +44,9 @@
 typedef unsigned int u32;
 typedef unsigned long long u64;
 
-static inline void base_writeb(u8 val, u32 offset)
+static inline void base_writel(u32 val, u32 offset)
 {
-	*((volatile u32 *)(offset)) = (u32)val;
+	*((volatile u32 *)(offset)) = val;
 }
 
 static inline u32 base_readl(u32 offset)
@@ -54,18 +54,6 @@
 	u32 reg = *((volatile u32 *)(HL_CFG_BASE + offset));
 
 	return reg;
-}
-
-static inline u8 regmap_readb(u32 offset)
-{
-	u32 reg = base_readl(HL_CFG_BASE + offset);
-
-	return (u8)(reg & 0xFF);
-}
-
-static inline void regmap_writeb(u8 val, u32 offset)
-{
-	base_writeb((u32)val, HL_CFG_BASE + offset);
 }
 
 static inline u32 regmap_readl(u32 offset)
@@ -77,7 +65,7 @@
 
 static inline void regmap_writel(u32 val, u32 offset)
 {
-	base_writeb(val, HL_CFG_BASE + offset);
+	base_writel(val, HL_CFG_BASE + offset);
 }
 
 static inline u32 rom_readl(u32 offset)
@@ -103,7 +91,7 @@
 
 static inline void fw_sram_writel(u32 val, u32 offset)
 {
-	base_writeb(val, HL_FWSRAM_BASE + offset);
+	base_writel(val, HL_FWSRAM_BASE + offset);
 }
 
 #endif /* _ARCBOOT_H_ */
